app/controllers/api/v1/rooms_controller.rb

```diff
@@ -114,6 +116,14 @@ def find_room
       def room_params
         params.require(:room).permit(:name, :user_id, :presentation)
       end
+
+      def room_limit_reached(user)
+        if RolePermission.joins(:permission).find_by(role_id: user.role_id, permission: { name: 'RoomLimit' }).value.to_i <= user.rooms.count.to_i
```

Comment:
This is starting to feel that it should be extracted in a `RolePermission` or a `Permission` class method,

Reply:
Did you think about adding this logic to services/PermissionsChecker.rb instead?

Reply:
I feel it would complicate things a bit to check for this specific case in the `PermissionsChecker`. `PermissionsChecker` i feel should stick to only returning whether or not the user has the permission.

Reply:
Decided to go with Amir's suggestion of adding the function to the `RolePermission` class

Reply:
I can help you with tomorrow if you want. I agree it's more complicated. I guess it's the price to pay for consistensy. On the top of my mind, I think it would work pretty much like every other permission.

Reply:
I agree with what Samuel suggested,
This is indeed requiring what the `call` logic is providing,
It's a permission,
That may remove the need to do that minor refactoring to provide a method that does the joining and fetches a permission record cleanly (if the same logic isn't used in several areas of the app),